assignments/arrays.js

```diff
@@ -63,34 +63,72 @@ let inventory = [{"id":1,"car_make":"Lincoln","car_model":"Navigator","car_year"
-console.log(`Car 33 is a *car year goes here* *car make goes here* *car model goes here*` );
-
 
+for (let i = 0; i < inventory.length; i++) {
+    if (inventory[i].id === 33) {
+        console.log(`Car 33 is a ${inventory[i].car_year} ${inventory[i].car_make} ${inventory[i].car_model}`);
+    }
+}
 
 // ==== Challenge 2 ====
 // The dealer needs the information on the last car in their inventory.  What is the make and model of the last car in the inventory?  Log the make and model into the console.
 let lastCar = 0;
-console.log();
+for (let i = 0; i < inventory.length; i++) {
+    lastCar = i;
+}
+console.log(`Last car is a ${inventory[lastCar].car_make} ${inventory[lastCar].car_model}`);
 
 // ==== Challenge 3 ====
 // The marketing team wants the car models listed alphabetically on the website. Sort all the car model names into alphabetical order and log the results in the console
 let carModels = [];
-console.log();
+
+for (let i = 0; i < inventory.length; i++) {
+    carModels.push(inventory[i].car_model.charAt(0).toUpperCase() + inventory[i].car_model.substr(1));
+}
+
+    //Sorting function
+function bubbleSort(arr){
+    let len = arr.length;
+    for (let i = len-1; i>=0; i--){
+      for(let j = 1; j<=i; j++){
+        if(arr[j-1]>arr[j]){
+            let temp = arr[j-1];
+            arr[j-1] = arr[j];
+            arr[j] = temp;
+         }
+      }
+    }
+    return arr;
+} //<---- function ends
+
+console.log(bubbleSort(carModels));
 
 // ==== Challenge 4 ====
 // The accounting team needs all the years from every car on the lot. Create a new array from the dealer data containing only the car years and log the result in the console.
 let carYears = [];
-console.log();
+for (let i = 0; i < inventory.length; i++) {
+    carYears.push(inventory[i].car_year);
+}
+console.log(carYears);
 
 // ==== Challenge 5 ====
 // The car lot manager needs to find out how many cars are older than the year 2000. Using the carYears array you just created, find out how many cars were made before the year 2000 by populating the array oldCars and logging it's length.
 let oldCars =[];
-console.log(); 
+for (let i = 0; i < inventory.length; i++) {
+    if (inventory[i].car_year < 2000) {
+        oldCars.push(inventory[i].car_year);
+    }
+}
+console.log(oldCars.length); 
 
 // ==== Challenge 6 ====
 // A buyer is interested in seeing only BMW and Audi cars within the inventory.  Return an array that only contains BMW and Audi cars.  Once you have populated the BMWAndAudi array, use JSON.stringify() to show the results of the array in the console.
 let BMWAndAudi =[];
-console.log();
-
-
-
+for (let i = 0; i < inventory.length; i++) {
+    let carMake = inventory[i].car_make;
+    if (carMake === "BMW" || carMake === "Audi") {
+        BMWAndAudi.push(inventory[i]);
+    }
+}
+var myJSON = JSON.stringify(BMWAndAudi);
```

Comment:
try to use const as much as possible as var will give you issues in a bigger code base.